portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,61 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.sps.data.DataStats;
+import com.google.gson.Gson;
+import java.util.Date;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+   private final Date dateCreated = new Date();
+   public ArrayList<String> jsonData = new ArrayList<String>();
```

Comment:
public final ArrayList<...

Also it's usually better to keep member variables private and only allow access via object methods

Reply:
Also HttpServlet typically runs in multi-threaded binaries (see https://docs.oracle.com/javaee/7/api/javax/servlet/http/HttpServlet.html), so your class needs to be thread safe. Java collections classes are not thread safe. You probably want to use one of the thread safe collections in https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/package-summary.html package